source/MapPanel.cpp

```diff
@@ -102,14 +102,18 @@ void MapPanel::Draw()
 	if(!distance.HasRoute(selectedSystem))
 	{
 		const Font &font = FontSet::Get(18);
-		
-		static const string NO_ROUTE = "You have not yet mapped a route to this system.";
 		Color black(0., 1.);
 		Color red(1., 0., 0., 1.);
-		Point point(-font.Width(NO_ROUTE) / 2, Screen::Top() + 40);
 		
-		font.Draw(NO_ROUTE, point + Point(1, 1), black);
-		font.Draw(NO_ROUTE, point, red);
+		string errormsg = "You have no";
+		if(player.HasSeen(selectedSystem))
```

Comment:
I had wanted to use `HasSeen` because then the player would get told about having no available route to a system that they had mapped (e.g. visited a neighbor at some point and thus knew a route to it), but hadn't fully considered the implication from the mission standpoint.

Rather than make a custom function just so the error message can be a touch more accurate, I switched to `HasVisited`. This way, only a system the player has actually routed to (and for some reason, can no longer route to anymore) will trigger the "You have no available route" message.

Fixed the style clash as well.